dataTransferScripts/register_users.py

```diff
@@ -0,0 +1,27 @@
+import os
+import firebase_admin
+from firebase_admin import credentials
+from firebase_admin import firestore
+from firebase_admin import auth
+import json
+
+########## Registers all users as firebase users in the firebase authentication part of our app, follow this by adding their additional data with the 'store_users.py' script ##############
+## I had to manually go through the Keystone data for users and delete users who had more than one account with the same email (firebase auth does not allow this). I kept the most recently created account in most cases
+project_id = 'cru-central-coast-prod'
+# Use the downloaded json file with app credentials
+cred = credentials.Certificate('/Users/jacobnogle/Desktop/CruApp/cru-central-coast-prod-firebase-adminsdk-yq1hp-b7c4990482.json')
+firebase_admin.initialize_app(cred)
+
+db = firestore.client()
+objects = []
+
+for line in open('users.json', 'r'):
+	objects.append(json.loads(line))
+
+for o in objects:
+	username = o['email']
+	password = o['password']
```

Comment:
So the one issue I see is that the password field you are getting is a hash of the password which means the users won't be able to log in. Also if users want to use another service to log in since that is now an option would they be able to switch that or are they now stuck using a local account?

Reply:
I guess I didn't realize the passwords were hashed. However, the script should still be the same if you are able to get a dump with the raw password data. To answer the question about federated login providers, users will be able to have multiple accounts using the same email with these providers (e.g. can login using Google, Facebook, or their local account using the same email). The only restriction is that they cannot create more than one local account using the same email, which is the issue I messaged you about earlier and why I had to delete some of the data before getting that script to work. I can send you the modified data file too if you would like.

Reply:
@fedorareis I'm not totally sure. I really wasn't able to find out much about federated logins. There was an option in the console to allow multiple accounts per email and I thought this would solve our issue with some people having multiple local accounts, but it still doesn't allow users to have multiple local accounts, only multiple federated accounts. This thread might help you understand better: https://support.google.com/firebase/answer/6400716